dspace-server-webapp/src/main/java/org/dspace/app/rest/repository/BundlePrimaryBitstreamLinkRepository.java

```diff
@@ -53,4 +65,92 @@ public BitstreamRest getPrimaryBitstream(@Nullable HttpServletRequest request,
             throw new RuntimeException(e);
         }
     }
+
+    /**
+     * Sets a primaryBitstream on a Bundle.
+     *
+     * @param context       The current DSpace context
+     * @param bundleId      The UUID of the Bundle
+     * @param bitstream     The Bitstream to use as primaryBitstream
+     * @param projection    The projection to use
+     * @return              The Bundle
+     */
+    @PreAuthorize("hasPermission(#bundleId, 'BUNDLE', 'WRITE')")
+    public BundleRest createPrimaryBitstream(Context context, UUID bundleId,
+                                             Bitstream bitstream, Projection projection) {
+        try {
+            Bundle bundle = setPrimaryBitstream(context, bundleId, bitstream, true);
+            return converter.toRest(context.reloadEntity(bundle), projection);
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Updates a primaryBitstream on a Bundle.
+     *
+     * @param context       The current DSpace context
+     * @param bundleId      The UUID of the Bundle
+     * @param bitstream     The Bitstream to use as primaryBitstream
+     * @param projection    The projection to use
+     * @return              The Bundle
+     */
+    @PreAuthorize("hasPermission(#bundleId, 'BUNDLE', 'WRITE')")
+    public BundleRest updatePrimaryBitstream(Context context, UUID bundleId,
+                                             Bitstream bitstream, Projection projection) {
+        try {
+            Bundle bundle = setPrimaryBitstream(context, bundleId, bitstream, false);
+            return converter.toRest(context.reloadEntity(bundle), projection);
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Deletes the primaryBitstream on a Bundle.
+     *
+     * @param context       The current DSpace context
+     * @param bundleId      The UUID of the Bundle
+     */
+    @PreAuthorize("hasPermission(#bundleId, 'BUNDLE', 'DELETE')")
```

Comment:
Again, this should be Bundle WRITE.  You aren't deleting anything, just unlinking the primaryBitstream.  We may want to clarify that in the JavaDocs here too... Delete in this situation is just making it so the Bundle has no primary bitstream.  The Bitstream is not deleted or removed from the Bundle, it's just no longer flagged as "primary".

Reply:
Updated similarly